### PR TITLE
fixes 2877: deletion of the organization or location in the context no longer breaks the application.

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by domcleal** [comment]
@isratrade, could you review please?

**Comment by domcleal** [comment]
Don't worry, I've reviewed.  This looks OK to me, could you add a test please?

**Comment by dmitri-d** [comment]
@domcleal: now with tests! and rainbows! and unicorns!

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -343,10 +343,11 @@ def set_taxonomy
       Organization.current = if orgs.count == 1 && !User.current.admin?
                                orgs.first
                              elsif session[:organization_id]
-                               orgs.find(session[:organization_id])
+                               orgs.find(session[:organization_id]) rescue nil
```
Comment (domcleal) [inline app/controllers/application_controller.rb]: How about find_by_id rather than catching an/all exception(s)?

Comment (dmitri-d) [inline app/controllers/application_controller.rb]: @domcleal: good idea

**Comment by dmitri-d** [comment]
@domcleal: ping?

**Comment by domcleal** [comment]
Merged as 2ac3855, thanks @witlessbird!

**Pull request closed without merging**